File: src/test/java/br/com/blz/testjava/dto/InventoryDtoTest.java
package br.com.blz.testjava.dto;

import br.com.blz.testjava.domain.Warehouse;
import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class InventoryDtoTest {

    @Test
    public void inventoryDtoQtd1Test() {
        List<Warehouse> warehouses = new ArrayList();
        Warehouse warehouse = new Warehouse();
        warehouse.setLocality("SP");
        warehouse.setQuantity(1);
        warehouse.setType("PHYSICAL_STORE");
        warehouses.add(warehouse);

        InventoryDto inventoryDto = new InventoryDto(warehouses);

        int quantityExpected = 1;

        assertEquals("Quantidade não compatível", quantityExpected, inventoryDto.getQuantity());
    }

    @Test
    public void inventoryDtoQtd6Test() {
        List<Warehouse> warehouses = new ArrayList();
        Warehouse warehouse = new Warehouse();
        warehouse.setLocality("SP");
        warehouse.setQuantity(1);
        warehouse.setType("PHYSICAL_STORE");
        warehouses.add(warehouse);

        warehouse = new Warehouse();
        warehouse.setLocality("SP");
        warehouse.setQuantity(5);
        warehouse.setType("PHYSICAL_STORE");
        warehouses.add(warehouse);

        InventoryDto inventoryDto = new InventoryDto(warehouses);

        int quantityExpected = 6;

        assertEquals("Quantidade não compatível", quantityExpected, inventoryDto.getQuantity());
    }

    @Test
    public void inventoryDtoQtd10Test() {
        List<Warehouse> warehouses = new ArrayList();
        Warehouse warehouse = new Warehouse();
        warehouse.setLocality("SP");
        warehouse.setQuantity(1);
        warehouse.setType("PHYSICAL_STORE");
        warehouses.add(warehouse);

        warehouse = new Warehouse();
        warehouse.setLocality("SP");
        warehouse.setQuantity(5);
        warehouse.setType("PHYSICAL_STORE");
        warehouses.add(warehouse);

        warehouse = new Warehouse();
        warehouse.setLocality("SP");
        warehouse.setQuantity(4);
        warehouse.setType("PHYSICAL_STORE");
        warehouses.add(warehouse);

        InventoryDto inventoryDto = new InventoryDto(warehouses);

        int quantityExpected = 10;

        assertEquals("Quantidade não compatível", quantityExpected, inventoryDto.getQuantity());
    }
}

File: src/test/kotlin/br/com/blz/testjava/controller/dto/InventoryDTOTest.kt
package br.com.blz.testjava.controller.dto

import br.com.blz.testjava.domain.entities.Inventory
import br.com.blz.testjava.domain.entities.Warehouse
import br.com.blz.testjava.domain.entities.WarehouseType
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class InventoryDTOTest {

  @Test
  fun `Must convert an Inventory with quantity 10`() {
    val warehouses = listOf(Warehouse("SP", 10, WarehouseType.ECOMMERCE))
    val inventory = Inventory(warehouses)

    val inventoryDTO = InventoryDTO(inventory)

    assertEquals(inventoryDTO.quantity, inventory.getQuantity())
    assertEquals(inventoryDTO.warehouses.size, inventory.warehouses.size)
  }

}
